Mathlib/AlgebraicGeometry/EllipticCurve/NormalForms.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.AlgebraicGeometry.EllipticCurve.Weierstrass
+
+/-!
+
+# Some normal forms of elliptic curves
+
+This file defines some normal forms of Weierstrass equations of elliptic curves.
+
+## Main definitions
+
+- ...
+
+## References
+
+* [J Silverman, *The Arithmetic of Elliptic Curves*][silverman2009]
+
+## Tags
+
+elliptic curve, weierstrass equation, normal form
+
+-/
+
+universe u
+
+variable {R : Type u}
+
+namespace WeierstrassCurve
+
+variable (W : WeierstrassCurve R)
+
+/-! ### Normal forms of characteristic ≠ 2
+-/
+
+/-- A `WeierstrassCurve` is of normal form of characteristic ≠ 2, if its a₁ and a₃ are zero.
+In other words it is $Y^2 = X^3 + a_2X^2 + a_4X + a_6$. -/
+@[mk_iff]
+structure IsNormalFormOfCharNeTwo [Zero R] : Prop where
```

Comment:
What do you think of abbreviating this name as follows:
```suggestion
structure IsNormalForm₂ [Zero R] : Prop where
```

Reply:
A question is that this is the normal form of characteristic not equal to 2. There are also normal forms of characteristic not equal to 2 or 3, characteristic equal to 2, and characteristic equal to 3. The `IsNormalForm₂` confuses the characteristic not equal to 2 case and characteristic equal to 2 case.

But I admit that this name is too long. Do you think `IsNFCharNeTwo` or `IsNFCharNe2` is OK?

Reply:
I like `IsNFCharNeTwo` better, but maybe `IsCharNeTwoNF` is more grammatical? Same goes for `IsCharTwoNF` and maybe `IsCharNeThreeNF` for char not 2 or 3 (unless you prefer `IsCharNeTwoThreeNF`)?

Reply:
I think `IsCharNeTwoNF` is good.